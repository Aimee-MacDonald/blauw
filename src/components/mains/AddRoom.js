import React from 'react'
import styled from 'styled-components'

import RoomsList from './booking_sheet/RoomsList'

const AddRoom = () => (
  <StyledAddRoom>
    <StyledEditSection>
      <form>
        <label>Name</label>
        <input/>

        <label>Group</label>
        <input/>

        <label>Max Pax</label>
        <input/>

        <button>Cancel</button>
        <button>Add Room</button>
      </form>
    </StyledEditSection>

    <RoomsList />
  </StyledAddRoom>
)

const StyledAddRoom = styled.div`
  display: grid;
  grid-template-columns: 64% 1fr;
  height: 100%;
`

const StyledEditSection = styled.div`
  display: grid;
  place-items: center center;

  form{
    display: grid;
    grid-template-columns: 1fr 1fr;
    width: 50%;
  }

  form label{
    text-align: right;
  }
`

export default AddRoom