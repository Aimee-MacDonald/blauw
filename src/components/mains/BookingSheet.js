import React from 'react'
import styled from 'styled-components'

import Dates from './booking_sheet/Dates'

const BookingSheet = () => (
  <StyledBookingSheet>
    <Dates/>
  </StyledBookingSheet>
)

const StyledBookingSheet = styled.div`
  background-color: indigo;
  width: 100%;
  height: 100%;
`

export default BookingSheet