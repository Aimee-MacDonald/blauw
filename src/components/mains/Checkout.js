import React from 'react'

export const Checkout = () => (
  <h1>Checkout</h1>
)

export default Checkout