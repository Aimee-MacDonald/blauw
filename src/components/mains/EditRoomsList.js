import React from 'react'

const EditRoomsList = () => <h1>Edit Rooms List</h1>

export default EditRoomsList