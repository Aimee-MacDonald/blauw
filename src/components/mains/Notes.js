import React from 'react'

export const Notes = () => (
  <h1>Notes</h1>
)

export default Notes