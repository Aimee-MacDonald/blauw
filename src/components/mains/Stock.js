import React from 'react'

export const Stock = () => (
  <h1>Stock</h1>
)

export default Stock