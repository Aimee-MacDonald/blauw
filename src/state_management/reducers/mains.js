const defaultState = {
  bookingSheet: {
    open: false
  },
  checkout: {
    open: false
  }
}

export default (state = defaultState, {type, main}) => {
  switch(type){
    case 'SET_MAIN':
      switch(main){
        case 'bookingSheet':
          return{
            ...defaultState,
            bookingSheet: {
              open: true
            }
          }

        case 'checkout':
          return{
            ...defaultState,
            checkout: {
              open: true
            }
          }
      }

    default:
      return state
  }
}