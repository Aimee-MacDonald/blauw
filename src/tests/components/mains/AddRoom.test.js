import React from 'react'
import {shallow} from 'enzyme'

import {AddRoom} from '../../../components/mains/AddRoom'

test('Render default Add Room screen', () => {
  const wrapper = shallow(<AddRoom/>)
  expect(wrapper.getElement()).toMatchSnapshot()
})