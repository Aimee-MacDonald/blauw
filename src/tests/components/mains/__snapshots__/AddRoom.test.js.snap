// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render default Add Room screen 1`] = `
<styled.div>
  <styled.div>
    <form>
      <label>
        Name
      </label>
      <input />
      <label>
        Group
      </label>
      <input />
      <label>
        Max Pax
      </label>
      <input />
      <button>
        Cancel
      </button>
      <button>
        Add Room
      </button>
    </form>
  </styled.div>
  <Memo(Connect(RoomsList)) />
</styled.div>
`;
